daphne/src/roles.rs

```diff
@@ -992,3 +992,28 @@ where
 
     Ok(())
 }
+
+/// Check for transition failures due to:
+///
+/// * the report having already been processed
+/// * the report having already been collected
+/// * the report not being within time bounds
+pub fn early_metadata_check(
+    metadata: &ReportMetadata,
+    processed: bool,
+    collected: bool,
+    min_time: u64,
```

Comment:
Suggestion: To make this method a bit more ergonomic, we might consider making `early_metadata_check()` a derived method on trait `DapAggregator` so that, instead of computing `min_time` and `max_time` externally, we could could have this method sample the clock and compute the bounds itself.

Would this work? Or is the computation of the time bounds Daphne-Worker specific? We might require `DapAggregator` to specify bounds for the window if time it'll accept reports.

Reply:
We probably could do that, and I considered it, but I went the way I did as I wanted easy, clock-independent unit testing of the key policy.  The current policy on computing time bounds, though implemented in the worker, could be moved to the Daphne crate as it only uses global configuration.  I'm not sure that's a good idea though, as one could imagine future time bounds policy having some worker-specific inputs, e.g. total storage used.